### PR TITLE
feat: add stats preprocessing functionality

### PR DESCRIPTION
**Pull request by brycetham**
Add the ability to preprocess/modify stats between getting the stats from the peer connection and the delta compression.

@types/node is required to use Promise.resolve().

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "build": "tsc rtcstats.ts --outDir dist --declaration"
   },
   "dependencies": {
+    "@types/node": "^18.7.6",
```
Comment (bbaldino) [inline package.json:20]: what required this?

Comment (brycetham) [reply package.json:20]: Promise.resolve(). See: https://heygabriel.medium.com/promise-only-refers-to-a-type-error-how-to-solve-it-ca41ce6cf691

File: rtcstats.ts [hunk 2]
```diff
@@ -106,12 +106,14 @@ const deltaCompression = (oldStats: StatsReport, newStats: StatsReport): StatsRe
...
  * @param id - Optional id string used for logging.
+ * @param statsPreProcessor - Optional function that modifies stats.
  */
 export const rtcStats = (
   pc: RTCPeerConnection,
   logger: (logData: LogData) => void,
   intervalTime: number,
-  id = v4()
+  id = v4(),
+  statsPreProcessor: (statsReport: StatsReport) => Promise<void> = () => Promise.resolve(),
```
Comment (bbaldino) [inline rtcstats.ts:116]: Can the default value here not be just a method reference to `Promise.resolve`?
```
statsPreProcessor: (statsReport: StatsReport) => Promise<void> = Promise.resolve,
```
does it complaint about a type mismatch there?

**Review by bbaldino — APPROVED** [review]
lgtm, one question but not blocking

File: rtcstats.ts (hunk 2 above)
Comment (brycetham) [reply rtcstats.ts:116]: It actually throws an error at runtime. JavaScript doesn't like assigning Promise.resolve that way: https://stackoverflow.com/questions/45210122/why-cant-promise-resolve-be-called-as-a-function

Comment (bbaldino) [reply rtcstats.ts:116]: Oh, interesting.  I think I've done this in a few places :grimacing:

**Pull request merged**